internal/discovery/dkv_discovery_client.go

```diff
@@ -0,0 +1,171 @@
+package discovery
+
+/*
+This class contains the behaviour of propagating a nodes status updates to discovery server
+ */
+
+import (
+	"context"
+	"fmt"
+	"github.com/flipkart-incubator/dkv/internal/hlc"
+	"github.com/flipkart-incubator/dkv/pkg/serverpb"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"gopkg.in/ini.v1"
+	"strconv"
+	"time"
+)
+
+type DiscoveryClientConfig struct {
+	discoveryServiceAddr 	string
+	// time in seconds to push status updates to discovery server
+	pushStatusInterval	 	time.Duration
+	// time in seconds to poll cluster info from discovery server
+	pollClusterInfoInterval	time.Duration
+}
+
+func NewDiscoveryClientConfigFromIni(sect *ini.Section) *DiscoveryClientConfig {
+	sectConf := sect.KeysHash()
+	if discoveryServiceAddr, ok := sectConf["discoveryServiceAddr"]; ok {
+		if pushStatusInterval, err := strconv.Atoi(sectConf["pushStatusInterval"]); err == nil {
+			if pollClusterInfoInterval, err := strconv.Atoi(sectConf["pollClusterInfoInterval"]); err == nil {
+				return NewDiscoveryClientConfig(discoveryServiceAddr, pushStatusInterval, pollClusterInfoInterval)
+			}
+		}
+	}
+	panic(fmt.Errorf("Invalid discovery client configuration. Check section %s", sect.Name()))
```

Comment:
Avoid throwing a panic, instead return a error so that caller can handle appropriately. A panic is equivalent to a fatal error that crashes that application.

Reply:
Fixed in #81